### ISSUE TITLE
Interleave training data from different labels prior to training.

### ISSUE THREAD

**Issue by john-philipp** [john-philipp/ml-impl-logistic-regression#7]
Avoid overfitting by interleaving data alternatingly prior to training on it.
